### PR TITLE
Feat/support all projects

### PR DESCRIPTION
**Pull request by mathild3r**
- [x] Tests written and linted [ℹ︎](https://github.com/snyk/general/wiki/Tests)
- [x] Documentation written [ℹ︎](https://github.com/snyk/general/wiki/Documentation)
- [ ] Commit history is tidy [ℹ︎](https://github.com/snyk/general/wiki/Git)

### What this does

_Explain why this PR exists_

add support for all-project 
https://www.notion.so/snyk/Support-for-all-project-126f5cd32b3c4d8abfd4dc088ad8c33c

### Notes for the reviewer

_Instructions on how to run this locally, background context, what to review, questions…_

### More information

- [Jira ticket SC-0000](https://snyksec.atlassian.net/browse/SC-0000)
- [Link to documentation](https://github.com/snyk/snyk-delta/wiki/)

### Screenshots

_Visuals that may help the reviewer_

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -39,7 +39,7 @@ or
     - Use the --setPassIfNoBaseline if used with snyk prevent commit status and the project is not monitored. This will prevent snyk-prevent_commit_status to fail.
     setPassIfNoBaseline default to false.\
      `snyk test --json --print-deps | snyk-delta --baselineOrg xxx --baselineProject xxx --setPassIfNoBaseline true`
-    - Use the --dontPassOnFail for snyk delta to fail if a new issues are found in a project. Default value: true
+    - Use the --passOnFail for snyk delta to fail if a new issues are found in a project. Default value: false
```
Comment (lili2311) [inline README.md:42]: this is a breaking change, your commits do not reflect that. We need to release a new major version. Is the breaking change intentional?

File: src/lib/error.ts [hunk 2]
```diff
@@ -3,32 +3,50 @@ import debugModule = require('debug');
...
             debug(error)
+            errorCode = 5
             break;
         case 'BadInputError':
             console.log("Bad input. Please check the --help")
             debug(error)
+            errorCode = 2
+            break;
+        case 'PrintDepsError':
+            console.log("print deps option detected")
```
Comment (lili2311) [inline src/lib/error.ts:35]: this error does not explain whats wrong, what does it mean?

File: README.md (hunk 1 above)
Comment (aarlaud) [reply README.md:42]: "passOnFail" and "snyk delta to fail if a new issues are found in a project" do not make sense. passOnFail true means it'll pass if new issues are found, passOnFail false means it'll fail if new issues are found. Is that right?

File: src/lib/error.ts (hunk 2 above)
Comment (aarlaud) [reply src/lib/error.ts:35]: Not clear on these codes either. Are error codes bringing something new that the console log doesn't bring already?

**Comment by mathild3r** [comment]
abandoned

**Pull request closed without merging**